### ISSUE TITLE
Agent in VF

### ISSUE THREAD

**Issue by fosterlynn** [valueflows/valueflows#423]
Not looking to past issues, too much to look at, and they would all be pretty old now.

We have been discussing how to document the pieces of other vocabularies that we include.  So I looked at org because they do that too (I don't know of others, most seem to just create a new version of the same or similar term).

The way org does it is they include the other vocab terms in their UML diagram.  They do not include them in their spec, but do include their relationships to them in their spec (makes sense).  They do make a list of other namespaces they want people to use to make their vocab complete in their view.

So, for skos:note it seems OK to me that we don't include that in our vocab, except documented on the UML diagram where it is useful and some doc recommending people use that.  Not super critical.

For Agent, that seems like another story since it is one of the R-E-A's - a pretty key concept.  (Agent in the org vocabulary is not as key as it is in REA, but more so than something like a note.)

So, one question: is foaf:Agent the same as the REA economic agent?  Then if it is, my question is can we really afford to not include some Agent in our spec?  Is the UML diagram plus a list enough?

**Comment by fosterlynn** [comment]
Part of this question is that org and the other vocabularies we are working with are pretty much only rdf-based vocabularies.  We want VF to be rdf core but also useful in any other format.  So we have more of a need to document for people who aren't used to, and aren't going to, figure out how to combine lots of vocabularies that are documented pretty much for LOD usage.

Maybe this just calls for some additional spec for non-rdf users?  Or maybe the UML diagram is enough?

**Comment by elf-pavlik** [comment]
I'll make PR to https://www.valueflo.ws/introduction/principles.html which adds, something like:
"VF only includes core terms specific to generalized economic activities. Developers using VF will need to augment it with other vocabularies and our documentation will recommend specific vocabs and terms to use."

> is foaf:Agent the same as the REA economic agent?

* foaf:Person - an individual
* org:Organization - an entity which 0 or more individuals can associate with (but distinct from any individual)

Does REA requires anything other than that?

Since you already have some collaboration with people working on ActivityPub implementation, maybe you could explore how you plan to relate to `as:Person`, `as:Group` and `as:Organization` ? I also wonder how those implementations treat `as:Group` and `as:Organization` differently and how instances of those two relate.

**Comment by pospi** [comment]
Does this include `AgentRelationship` stuff? I can't find any open issues about it, but from the comments @fosterlynn made on `vf-graphql` it seems like something that is going to be deprecated. (Should I create a new issue & move this?)

My thoughts:

- deprecate `agent.roles` & `AgentRelationshipRole`
- change `AgentRelationship` to replace the old `AgentRelationshipRole` field with a simple `classifiedAs`
- find an ontology that has `LEGALPARTNER`, `MEMBER` & other current defaults & recommend such as the default
- rename `memberRelationships` to simply `relationships`

**Comment by fosterlynn** [comment]
>Does this include AgentRelationship stuff? I can't find any open issues about it, but from the comments @fosterlynn made on vf-graphql it seems like something that is going to be deprecated. (Should I create a new issue & move this?)

@pospi No, nothing is being deprecated in the agent model, sorry if I confused things somehow.

This issue was more about how do we document things that are part of REA and will be used as part of this vocabulary, but for which we are re-using terms from other vocabularies.  We decided to leave them out of the rdf turtle spec, because that more properly would recognize just things in the VF namespace in Linked Open Data.  But to leave them in the other treatments and in the UML diagram so people are clear that they are actually part of the vocabulary that someone would require for VF economic activity. (That might explain why almost nobody actually re-uses other vocabularies!)  And they are also here https://valueflo.ws/specification/external-terms.html.

>deprecate agent.roles & AgentRelationshipRole

We have found consistently that people like to define their own roles, or at least rename 'standard' roles.  There also may end up being more configuration wanted as part of that, so I would expect it rather to grow some more properties.

>find an ontology that has LEGALPARTNER, MEMBER & other current defaults & recommend such as the default

We looked around quite a bit.  We followed ActivityStreams (I think this is what I mentioned) to sync up with the naming, but they don't have user defined roles.

>rename memberRelationships to simply relationships (edit: actually, just delete- relationships already exists)

We don't really need memberRelationships in the graphql treatment.  That was to make it easy for apps that mostly use "member" roles in OCP.  We do need to keep all the basic pieces of the relationships, with enough parameters to filter by role or eventually the mythical role behaviors.
